projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -114,6 +109,14 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   ngOnInit(): void {
+    /* this is needed to sync validation when the parent form is first submitted
+       without having any values in the sort code inputs */
+    this.invalidSortCode$?.pipe(take(1))?.subscribe(() => {
+      this.sortCodeFormGroup.markAllAsTouched();
+      Object.keys(this.sortCodeFormGroup.controls).forEach((fieldName) => {
+        this.sortCodeFormGroup.controls[fieldName].markAsDirty();
+      });
+    });
```

Comment:
Just wondering if there's a way to cover this in the unit tests? 🤔

Reply:
I will think of some tests to add 😎 👍

Reply:
@Zimann can you please write some unit tests?

Reply:
We could have a conversation about where to place these tests but for now, I'd say we let this one roll forward without them. Is that alright? 😄

Reply:
Unit tests tend to be super important for us, especially considering that Canopy is a package that's used in several applications. I would suggest that we give it a go now, and add some tests which mock the `statusChanges` observable to check that certain things have happened - e.g. that the controls are marked as dirty.

Reply:
I agree with @owensgit. We don't ship any code without unit tests ;) @Zimann I'm happy to help with the tests if needed.